rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java

```diff
@@ -248,47 +251,69 @@ public List<ModuleDescription> getRpcModules() {
+            List<? extends ConfigObject> list = configFromFiles.getObjectList(RPC_MODULES_PATH);
+            modules = getModulesFromListFormat(list);
+        } else {
+            ConfigObject configObject = configFromFiles.getObject(RPC_MODULES_PATH);
+            modules = getModulesFromObjectFormat(configObject);
+        }
+        this.moduleDescriptions = modules;
+        return modules;
+    }
 
+    private List<ModuleDescription> getModulesFromObjectFormat(ConfigObject modulesConfigObject) {
+        List<ModuleDescription> modules = new ArrayList<>();
+
+        for (String configKey : modulesConfigObject.keySet()) {
+            String name = configKey;
+            Config configElement = ((ConfigObject) modulesConfigObject.get(configKey)).toConfig();
+            modules.add(getModule(configElement, name));
+        }
+        return modules;
+    }
+
+    private List<ModuleDescription> getModulesFromListFormat(List<? extends ConfigObject> list) {
+        List<ModuleDescription> modules = new ArrayList<>();
         for (ConfigObject configObject : list) {
             Config configElement = configObject.toConfig();
             String name = configElement.getString("name");
-            String version = configElement.getString("version");
-            boolean enabled = configElement.getBoolean("enabled");
-            long timeout = 0;
-            Map<String, Long> methodTimeoutMap = new HashMap<>();
-
-            if (configElement.hasPath("timeout")) {
-                timeout = configElement.getLong("timeout");
-            }
-
-            if (configElement.hasPath("methods.timeout")) {
-                fetchMethodTimeout(configElement, methodTimeoutMap);
-            }
+            modules.add(getModule(configElement, name));
+        }
+        return modules;
+    }
+    private ModuleDescription getModule(Config configElement, String name) {
 
-            List<String> enabledMethods = null;
-            List<String> disabledMethods = null;
+        String version = configElement.getString("version");
+        boolean enabled = configElement.getBoolean("enabled");
+        List<String> enabledMethods = null;
+        List<String> disabledMethods = null;
+        int timeout = 0;
+        Map<String, Long> methodTimeoutMap = new HashMap<>();
 
-            if (configElement.hasPath("methods.enabled")) {
-                enabledMethods = configElement.getStringList("methods.enabled");
-            }
 
-            if (configElement.hasPath("methods.disabled")) {
-                disabledMethods = configElement.getStringList("methods.disabled");
-            }
+        if (configElement.hasPath("timeout")) {
+            timeout = configElement.getInt("timeout");
+        }
 
-            modules.add(new ModuleDescription(name, version, enabled, enabledMethods, disabledMethods, timeout, methodTimeoutMap));
+        if (configElement.hasPath("methodTimeout")) {
```

Comment:
shouldn't it be `"methods.timeout"` ? :) please cover this field in the tests

Reply:
yep, I'll add it to the tests.